src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -1178,7 +1185,19 @@ public void visitSources(SCMSourceObserver observer) throws IOException, Interru
                 }
                 if (user != null && repoOwner.equalsIgnoreCase(user.getLogin())) {
                     listener.getLogger().format("Looking up repositories of user %s%n%n", repoOwner);
-                    for (GHRepository repo : user.listRepositories(100)) {
+                    PagedIterable<GHRepository> repositories;
+                    if (githubAppAuthentication) {
+                        // If we get here, then the app is installed in a user's "organization", because
+                        // GET /org/:org returned null. GET /users/:user/repos will only include public
+                        // repositories, so we use an alternate API to list repositories available to the
+                        // installation instead.
+                        // https://docs.github.com/en/rest/apps/installations#list-repositories-accessible-to-the-app-installation
+                        repositories =
+                                github.getInstallation().listRepositories().withPageSize(100);
+                    } else {
+                        repositories = user.listRepositories(100);
+                    }
+                    for (GHRepository repo : repositories) {
```

Comment:
These changes are not directly related to the main goal of this change, but without them it is awkward to test GitHub App credentials unless you create a GitHub org for testing. With this change you can use app credentials with your own user account.